File: Núcleo/src/LogicaNegocio/Coordinador.java
package LogicaNegocio;

import Recursos.DatosPersonal;
import java.util.Date;

/**
 * @author Victor Javier
 * @version 1.0
 * @created 18-may.-2019 06:27:02 p. m.
 */
public class Coordinador extends Personal {
    private ICoordinador iCoordinador;
    
        
    public Coordinador() {

    }
    
    private DatosPersonal validarDatosPersonalesCoordinador(){
        DatosPersonal datosRecepcionista = DatosPersonal.VALIDO;
        if(this.getNumeroTelefono().trim().equals("") || this.getNumeroTelefono() == null){
            datosRecepcionista = DatosPersonal.NUMERO_TELEFONO_VACIO;
        }else if(this.getNumeroTelefono().trim().length() > 10){
            datosRecepcionista = DatosPersonal.NUMERO_TELEFONO_LARGO;
        }else if(this.getNumeroTelefono().trim().length() < 10){
            datosRecepcionista = DatosPersonal.NUMERO_TELEFONO_CORTO;
        }else if(this.getNombre().trim().equals("") || this.getNombre() == null){
            datosRecepcionista = DatosPersonal.NOMBRE_VACIO;
        }else if(this.getNombre().trim().length() > 70){
            datosRecepcionista = DatosPersonal.NOMBRE_LARGO;
        }else if(this.getApellido().trim().equals("") || this.getApellido() == null){
            datosRecepcionista = DatosPersonal.APELLIDO_VACIO;
        }else if(this.getApellido().trim().length() > 70){
            datosRecepcionista = DatosPersonal.APELLIDO_LARGO;
        }else if(this.getFechaNacimiento().toString().trim().equals("") || this.getFechaNacimiento() == null){
            datosRecepcionista = DatosPersonal.FECHA_VACIA;
        }else if(this.getFechaNacimiento() == new Date()){
            datosRecepcionista = DatosPersonal.FECHA_ACTUAL;
        }if(this.getSexo()==' '){
            datosRecepcionista = DatosPersonal.SEXO_VACIO;
        }else  if(this.getRfc().trim().equals("") || this.getRfc() == null){
            datosRecepcionista = DatosPersonal.RFC_VACIO;
        }else if(this.getRfc().trim().length() > 13){
            datosRecepcionista = DatosPersonal.RFC_LARGO;
        }else if(this.getRfc().trim().length()< 13){
            datosRecepcionista = DatosPersonal.RFC_CORTO;
        }else if(this.getNumeroPersonal().trim().equals("") || this.getNumeroPersonal() == null){
            datosRecepcionista = DatosPersonal.NUMERO_PERSONAL_VACIO;
        }else if(this.getTurno().trim().equals("") || this.getTurno() == null){
            datosRecepcionista = DatosPersonal.TURNO_VACIO;
        }
        return datosRecepcionista;
    }
    public void setICoordinador(ICoordinador iCoordinador){
        this.iCoordinador = iCoordinador;
    }
    
    private DatosPersonal validarDatosCoordinador(){
        DatosPersonal datosCoordinador = DatosPersonal.VALIDO;
        if(this.validarDatosPersonalesCoordinador() != DatosPersonal.VALIDO){
            datosCoordinador = this.validarDatosPersonalesCoordinador();
        }
        return datosCoordinador;
    }

    @Override
    public DatosPersonal registrar() {
        DatosPersonal coordinadorRegistrado = this.validarDatosCoordinador();
        if(coordinadorRegistrado == DatosPersonal.VALIDO){
            if(this.obtenerPersonal(this.getNumeroPersonal())== null){
                coordinadorRegistrado = this.iCoordinador.registrar(this)?
                        DatosPersonal.EXITO : DatosPersonal.ERROR_ALMACENAMIENTO;
            }
        }
        return coordinadorRegistrado;
    }

    @Override
    public DatosPersonal modificar() {
        DatosPersonal coordinadorModificado = this.validarDatosCoordinador();
        if(coordinadorModificado == DatosPersonal.VALIDO){
            coordinadorModificado = this.iCoordinador.modificar(this)?
                    DatosPersonal.EXITO : DatosPersonal.ERROR_ALMACENAMIENTO;
        }
        return coordinadorModificado;
    }

    @Override
    public boolean registrarEntrada(String numeroConsultorio) {
         return this.iCoordinador.registrarEntrada(this.getNumeroPersonal(), numeroConsultorio);
    }

    @Override
    public boolean registrarSalida() {
        return this.iCoordinador.registrarSalida(this.getNumeroPersonal());
       
    }

    @Override
    public boolean eliminar() {
        return this.iCoordinador.eliminar(this.getNumeroPersonal());
    }

    @Override
    public Personal obtenerPersonal(String numeroPersonal) {
        return this.iCoordinador.obtenerCoordinador(numeroPersonal);
    }
  
}//end Coordinador

File: Nucleo servidor web/application/libraries/Coordinador.php
<?php

if ( ! defined('BASEPATH')) exit('No direct script access allowed');
require_once(APPPATH.'libraries/recursos/DatosPersonal.php');
require_once(APPPATH.'libraries/Personal.php');

class Coordinador extends Personal{
    private $iCoordinador;

    public function __get($attr) {
		return CI_Controller::get_instance()->$attr;
    }
    
    public function __construct(){
        
    }

    private function validarDatosPersonalesCoordinador(){
        $datosCoordinador = DatosPersonal::VALIDO;
        if($this->getNumeroTelefono() === "" || $this->getNumeroTelefono() === null){
            $datosCoordinador = DatosPersonal::NUMERO_TELEFONO_VACIO;
        }else if(strlen($this->getNumeroTelefono()) > 10){
            $datosCoordinador = DatosPersonal::NUMERO_TELEFONO_LARGO;
        }else if(strlen($this->getNumeroTelefono()) < 10){
            $datosCoordinador = DatosPersonal::UMERO_TELEFONO_CORTO;
        }else if($this->getNombre() === "" || $this->getNombre() === null){
            $datosCoordinador = DatosPersonal::NOMBRE_VACIO;
        }else if(strlen($this->getNombre()) > 70){
            $datosCoordinador = DatosPersonal::NOMBRE_LARGO;
        }else if($this->getApellido()==="" || $this->getApellido() === null){
            $datosCoordinador = DatosPersonal::APELLIDO_VACIO;
        }else if(strlen($this->getApellido()) > 70){
            $datosCoordinador = DatosPersonal::APELLIDO_LARGO;
        }else if($this->getFechaNacimiento() === "" || $this->getFechaNacimiento() === null){
            $datosCoordinador = DatosPersonal::FECHA_VACIA;
        }else if($this->getSexo()==""){
            $datosCoordinador = DatosPersonal::SEXO_VACIO;
        }else  if($this->getRfc()==="" || $this->getRfc() === null){
            $datosCoordinador = DatosPersonal::RFC_VACIO;
        }else if(strlen($this->getRfc()) > 13){
            $datosCoordinador = DatosPersonal::RFC_LARGO;
        }else if(strlen($this->getRfc())< 13){
            $datosCoordinador = DatosPersonal::RFC_CORTO;
        }else if($this->getNumeroPersonal()==="" || $this->getNumeroPersonal() === null){
            $datosCoordinador = DatosPersonal::NUMERO_PERSONAL_VACIO;
        }else if($this->getTurno()==="" || $this->getTurno() == null){
            $datosCoordinador = DatosPersonal::TURNO_VACIO;
        }
        return $datosCoordinador;
    }
    public function setICoordinador($iCoordinador){
        $this->iCoordinador = $iCoordinador;
    }


    public function registrar($idUsuario){
        $coordinadorRegistrado = $this->validarDatosPersonalesCoordinador();
        if($coordinadorRegistrado == DatosPersonal::VALIDO){
            if($this->obtenerPersonal($this->getNumeroPersonal()) === null){
                if($this->iCoordinador->registrar($this, $idUsuario)){
                    $coordinadorRegistrado = DatosPersonal::EXITO;
                }else{
                    $coordinadorRegistrado =  DatosPersonal::ERROR_ALMACENAMIENTO;
                }
            }
        }
        return $coordinadorRegistrado;
    }
    public function modificar(){
        $coordinadorModificado = $this->validarDatosPersonalesCoordinador();
        if($coordinadorModificado === DatosPersonal::VALIDO){
            if($this->iCoordinador->modificar($this)){
                $coordinadorModificado = DatosPersonal::EXITO;
            }else{
                $coordinadorModificado = DatosPersonal::ERROR_ALMACENAMIENTO;
            }
        }
        return $coordinadorModificado;
    }
    public function registrarEntrada($numeroConsultorio){
        return $this->iCoordinador->registrarEntrada($this->getRfc(),$numeroConsultorio);
    }
    public function registrarSalida(){
        return $this->iCoordinador->registrarSalida($this->getRfc());
    }
    public function eliminar(){
        return $this->iCoordinador->eliminar($this->getRfc());
    }
    public function obtenerPersonal($rfc){
        return $this->iCoordinador->obtenerCoordinador($rfc);
    }
        public function obtenerPersonalId($idUsuario){
        return $this->iCoordinador->obtenerCoordinadorId($idUsuario);
    }
}